bin/check-ports-bind.rb

```diff
@@ -105,53 +96,101 @@ def severity(warn, text)
     end
   end
 
-  # Check address:port
-  def check_port(portbind, okays)
-    address = portbind.split(':')[0]
-    port = portbind.split(':')[1]
+  # Ports to check
+  def portbinds
+    binds = []
+
+    config[:portbinds].split(',').each do |portbind|
+      portbind = "#{config[:host]}:#{portbind}" unless portbind.include?(':')
+      portbind = "#{portbind}/tcp" unless portbind.include?('/')
+
+      protocol     = portbind.split('/')[1] || 'tcp'
+      address_port = portbind.split('/')[0]
+      address      = address_port.split(':')[0]
+      port         = address_port.split(':')[1].to_i
+
+      if port =~ /^[0-9]+(-[0-9]+)$/
```

Comment:
we should also check that they are actually between 0-65k while we are at it. I'd vote for a function called `valid_port?(port)`

Reply:
Great idea. What do you think about returning a critical result if this validation fails?